test/export/test_sparse.py

```diff
@@ -0,0 +1,229 @@
+            self.assertEqual(x._indices(), y._indices(), exact_layout=True)
+            self.assertEqual(x._values(), y._values(), exact_layout=True)
+        else:
+            if x.layout in {torch.sparse_csr, torch.sparse_bsr}:
+                x_meta1, y_meta1 = (x.crow_indices(), y.crow_indices())
+                x_meta2, y_meta2 = (x.col_indices(), y.col_indices())
+            elif x.layout in {torch.sparse_csc, torch.sparse_bsc}:
+                x_meta1, y_meta1 = (x.ccol_indices(), y.ccol_indices())
+                x_meta2, y_meta2 = (x.row_indices(), y.row_indices())
+            else:
+                self.assertFail()
+            self.assertEqual(x_meta1, y_meta1, exact_layout=True)
+            self.assertEqual(x_meta2, y_meta2, exact_layout=True)
+            self.assertEqual(x.values(), y.values(), exact_layout=True)
+
+    @unittest.skipIf(
+        sys.version_info >= (3, 12), "torch.compile is not supported on python 3.12+"
+    )
+    @parametrize("dtype", DTYPES)
+    @parametrize("itype", ITYPES)
+    @all_sparse_layouts("layout")
+    def test_idnet(self, dtype, itype, layout):
+        if layout is not torch.sparse_coo:
+            self.skipTest("TODO: support non-coo sparsity!")
+
+        net = IdNet()
+        for sparse_input in self.generate_simple_inputs(
+            layout,
+            device="cpu",
+            dtype=dtype,
+            index_dtype=itype,
+        ):
+            # Build the traced graph.
+            prog = torch.export.export(net, (sparse_input,))
+            # Test arg/output.
+            for i, node in enumerate(prog.graph.nodes):
+                meta = node.meta.get("val", None)
+                if i == 0:
+                    self.assertEqualMeta(meta, sparse_input)
+                else:
+                    self.assertEqual(meta, None)
+
+    @unittest.skipIf(
+        sys.version_info >= (3, 12), "torch.compile is not supported on python 3.12+"
+    )
+    @parametrize("dtype", DTYPES)
+    @parametrize("itype", ITYPES)
+    @all_sparse_layouts("layout")
+    def test_sumnet(self, dtype, itype, layout):
+        if layout is not torch.sparse_coo:
+            self.skipTest("TODO: support non-coo sparsity!")
+
+        net = SumNet()
+        for sparse_input in self.generate_simple_inputs(
+            layout,
+            device="cpu",
+            dtype=dtype,
+            index_dtype=itype,
+        ):
+            # Build the traced graph.
+            prog = torch.export.export(net, (sparse_input,))
+            # Test arg/sum/output.
+            for i, node in enumerate(prog.graph.nodes):
+                meta = node.meta.get("val", None)
+                if i == 0:
+                    self.assertEqualMeta(meta, sparse_input)
+                elif i == 1:
+                    self.assertIsInstance(meta, FakeTensor)
+                    self.assertEqual(meta.layout, torch.strided)
+                    self.assertEqual(meta.dtype, dtype)
```

Comment:
```suggestion
                    self.assertEqualMeta(meta, net.forward(sparse_input))
```
after applying
```
diff --git a/test/export/test_sparse.py b/test/export/test_sparse.py
index 1a64e6dc05f..5e332927b46 100644
--- a/test/export/test_sparse.py
+++ b/test/export/test_sparse.py
@@ -101,7 +101,9 @@ class TestSparseProp(TestCase):
         # comparing their values. In the case of sparse tensors, this means
         # that comparing indices and values attributes are skipped as well,
         # which is why we are doing that explicitly below.
-        if x.layout is torch.sparse_coo:
+        if x.layout is torch.strided:
+            pass
+        elif x.layout is torch.sparse_coo:
             self.assertEqual(x._indices(), y._indices(), exact_layout=True)
             self.assertEqual(x._values(), y._values(), exact_layout=True)
         else:
```

This suggestion ensures that also the shape of `meta` is checked. The shape is defined by that `net.forward` result that usage allows the generalization of the test to any net instance (that uses a similar pattern to SumNet).